azure/scope/managedcontrolplane_test.go

```diff
@@ -1780,3 +1780,98 @@ func TestManagedControlPlaneScope_GroupSpecs(t *testing.T) {
 		})
 	}
 }
+
+func TestManagedControlPlaneScope_EnablePreviewFeatures(t *testing.T) {
+	cases := []struct {
+		name     string
+		input    ManagedControlPlaneScopeParams
+		expected bool
+	}{
+		{
+			name: "Without EnablePreviewFeatures",
+			input: ManagedControlPlaneScopeParams{
+				Cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cluster1",
+						Namespace: "default",
+					},
+				},
+				ControlPlane: &infrav1.AzureManagedControlPlane{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cluster1",
+						Namespace: "default",
+					},
+					Spec: infrav1.AzureManagedControlPlaneSpec{
+						AzureManagedControlPlaneClassSpec: infrav1.AzureManagedControlPlaneClassSpec{
+							SubscriptionID: "00000000-0000-0000-0000-000000000000",
+						},
+					},
+				},
+			},
+			expected: false,
+		},
+		{
+			name: "With EnablePreviewFeatures false",
+			input: ManagedControlPlaneScopeParams{
+				Cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cluster1",
+						Namespace: "default",
+					},
+				},
+				ControlPlane: &infrav1.AzureManagedControlPlane{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cluster1",
+						Namespace: "default",
+					},
+					Spec: infrav1.AzureManagedControlPlaneSpec{
+						AzureManagedControlPlaneClassSpec: infrav1.AzureManagedControlPlaneClassSpec{
+							SubscriptionID:        "00000000-0000-0000-0000-000000000000",
+							EnablePreviewFeatures: ptr.To(false),
+						},
+					},
+				},
+			},
+			expected: false,
+		},
+		{
+			name: "With EnablePreviewFeatures true",
+			input: ManagedControlPlaneScopeParams{
+				Cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cluster1",
+						Namespace: "default",
+					},
+				},
+				ControlPlane: &infrav1.AzureManagedControlPlane{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cluster1",
+						Namespace: "default",
+					},
+					Spec: infrav1.AzureManagedControlPlaneSpec{
+						AzureManagedControlPlaneClassSpec: infrav1.AzureManagedControlPlaneClassSpec{
+							SubscriptionID:        "00000000-0000-0000-0000-000000000000",
+							EnablePreviewFeatures: ptr.To(true),
+						},
+					},
+				},
+			},
+			expected: true,
+		},
```

Comment:
This is a massive test case definition to test a one-line method call. I would not lose any sleep if we don't test this method at all.